docstore/awsdynamodb/dynamo.go

```diff
@@ -131,12 +132,11 @@ func newCollection(db *dyn.DynamoDB, tableName, partitionKey, sortKey string, op
 
 // Key returns a two-element array with the partition key and sort key, if any.
 func (c *collection) Key(doc driver.Document) (interface{}, error) {
-	var keys [2]interface{}
-	var err error
-	keys[0], err = doc.GetField(c.partitionKey)
-	if err != nil {
+	pkey, err := doc.GetField(c.partitionKey)
+	if err != nil || pkey == nil || driver.IsEmptyValue(reflect.ValueOf(pkey)) {
```

Comment:
I've tried in `IsEmptyValue` to loop through an array to check whether each element is also empty, but that didn't work for a case like `[2]interface{}{"", nil}`, since both elements are `reflect.Interface` kind and the first one is not a "zero interface". Another reason is that only the partition key is actually required.

Reply:
What you have seems fine.